scripts/up.sh

```diff
@@ -1,79 +1,54 @@
 #!/usr/bin/env bash
-#
-# Bring up the SDK test environment.
 
-set -e
+set -ae
+START=$(date "+%s")
 
-rootdir=`dirname $0`
-pushd $rootdir/.. > /dev/null
 
-# Defaults
-TYPE_OVERRIDE=""
-DAEMON_FLAG="-d"
-SKIP_BUILD=0
-SHOW_HELP=0
-ENV_FILE=".up-env"
-PARALLEL="--parallel"
+ENV_FILE=".env"
+echo "up.sh: sourcing environment vars from-->$ENV_FILE"
+source "$ENV_FILE"
 
-show_help() {
-  echo "Manage bringing up the SDK test environment."
-  echo
-  echo "Usage: up.sh [options]"
-  echo
-  echo "Options:"
-  echo "  -f <FILE>  Override the environment file."
-  echo "  -t <TYPE>  Override the installation type specified in the environment file."
-  echo "             Valid types: ['channel', 'type']"
-  echo "  -s         Skip rebuilding the docker image."
-  echo "  -i         Start the docker environment in interactive mode."
-  echo "  -h         Provide this help information."
-  echo "  -p         Disable parallel build mode. Probably only useful when editing the test environment."
-}
-
-# Parse arguments
-while getopts "pf:t:sih" opt; do
-  case "$opt" in
-    f) ENV_FILE=$OPTARG; ;;
-    i) unset DAEMON_FLAG; ;;
-    s) SKIP_BUILD=1; ;;
-    t) TYPE_OVERRIDE=$OPTARG; ;;
-    h) show_help; exit 0 ;;
-    p) unset PARALLEL; ;; 
-  esac
-done
-
-# Verify there are no positional parameters with getopt/getopts
-shift "$((OPTIND-1))"
-if [[ "$1" != "" ]]; then
-  echo "No positional arguments should be provided, found '$@'"
-  echo
-  show_help
-  exit
-fi
-
-# Load environment.
-source $ENV_FILE
-
-# Choose which dockerfile to use.
-TYPE=${TYPE_OVERRIDE:-$TYPE}
 if [[ $TYPE == "channel" ]] || [[ $TYPE == "source" ]]; then
   export TYPE="$TYPE"
+  if [[ $TYPE == "channel" ]]; then
+    ALGOD_URL=""
+    ALGOD_BRANCH=""
+    ALGOD_SHA=""
+  else
+    ALGOD_CHANNEL=""
+  fi
 else
   echo "Unknown environment: $TYPE"
   exit 1
 fi
 
-echo "Bringing up network with '$TYPE' configuration."
-
-# Make sure it isn't running
+# Make sure test-sdk sandbox isn't running and clean up any docker detritous
+echo "up.sh: Before bootstrapping, try cleaning up first..."
 ./scripts/down.sh
+rm -rf "$LOCAL_SANDBOX_DIR"
+echo "up.sh: seconds it took to get to end of $LOCAL_SANDBOX_DIR cleanup: " + $(($(date "+%s") - $START))
 
-# Remove the containers to allow re-running stateful tests.
-docker-compose rm --force
 
-# When developing, it's often useful to skip the build phase.
-if [[ $SKIP_BUILD -eq 0 ]]; then
-  docker-compose build --no-cache $PARALLEL
-fi
+rootdir=$(dirname "$0")
+pushd "$rootdir"/.. > /dev/null || exit
+
+git clone --branch "$SANDBOX_BRANCH" --single-branch "$SANDBOX_URL" "$LOCAL_SANDBOX_DIR"
```

Comment:
@tzaffi Minor and optional:  To speed up git clone operations, it's possible to clone _only_ the most recent commit (source: https://stackoverflow.com/a/1210012).

sandbox is probably a smaller repo, so treat optionally.  If nothing else, I see a benefit because the clone statement is liable to be copy-pasted later on.

```suggestion
git clone --depth 1 --branch "$SANDBOX_BRANCH" --single-branch "$SANDBOX_URL" "$LOCAL_SANDBOX_DIR"
```

Reply:
nice!